### PR TITLE
[TASK] TYPO3 v12 compatibility

### PR DESCRIPTION
**Pull request by ervaude**

### CODE REVIEW

File: Classes/Controller/Ajax/TranslationController.php [hunk 1]
```diff
@@ -46,33 +47,33 @@ public function __construct(
...
+            $this->translationFileService->updateSourceInFiles($postParams);
+        }
+        $this->flushCache();
+        $content = [
+            'flashMessage' => [
+                'title' => 'OK',
+                'message' => 'label updated',
+                'severity' => AbstractMessage::OK,
+            ],
+        ];
```
Comment (achimfritz) [inline Classes/Controller/Ajax/TranslationController.php]: should this not be inside the try/catch block? (try block is empty?)

File: Classes/Domain/Model/TranslationFile.php [hunk 2]
```diff
@@ -13,25 +14,32 @@
...
+        // Escape directory separators in path to vendor directory
+        $pathPart = str_replace('/', '\/', Environment::getProjectPath() . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR);
+        // Remove the path to the vendor directory from the path to the file
         $this->relativePath = preg_replace('/' . $pathPart . '/', '', $this->getCleanPath());
+        // Find the vendor/package-name used by composer
+        $matches = [];
+        preg_match('#^[^\/]*\/[^\/]*#', $this->relativePath, $matches);
+        // Convert the composer package name into the TYPO3 extension name.
+        // Append the extension name instead of the composer package name to the relative path.
+        // Example: Instead of "vendor-name/my-extension/Resources/Private/" we will get "my_extension/Resources/Private/",
```
Comment (achimfritz) [inline Classes/Domain/Model/TranslationFile.php:39]: what will happen if package key is foo/barBaz ? and extensionKey is bar_baz?
cant we get die EXT-Key from `Package->getComposerManifest('vendor-name/my-extension')`?

Comment (ervaude) [reply Classes/Domain/Model/TranslationFile.php:39]: that would still work

File: Classes/Controller/Ajax/TranslationController.php (hunk 1 above)
Comment (ervaude) [reply Classes/Controller/Ajax/TranslationController.php]: ah thanks, I pulled it out of the try block to see the exception :)

**Review by achimfritz — APPROVED** [review]

**Pull request merged**